### DISCUSSION TITLE
A Suggestion that could really amp up our project with Gemini 1.5 pro's large Context Windows

### DISCUSSION THREAD

**Discussion by rajofearth** [Pythagora-io/gpt-pilot#719] [Ideas]
---

**Hey GPT Pilot Team,**

I've got a suggestion that could really amp up our project. What if we added support for Gemini models like Gemini 1.0 Pro and[ Gemini 1.5 Pro](https://ai.google.dev/)?
Think about the possibilities – developers everywhere getting even more out of GPT Pilot with [Gemini models](https://ai.google.dev/docs) in the mix with its large content windows. It's an exciting prospect that could truly elevate our project.

So, what do you think? Let's explore this idea and see where it takes us!

---

**Comment by ssnaikr** [comment]
Not only Gemini 1.5 Pro but also Claude 3.
I understand they support azure for open source models, but it should be on evreryones interest to be able to use more than one of the top closed source leaders.

**Reply by ssnaikr** [reply]
I see that @nimrod4278 is working on something.
But something for Claude 3 would also benefit.
In this same thread someone talks about a plug gable interface so we can use any model. That would be the best way to go.
https://github.com/Pythagora-io/gpt-pilot/pull/512

**Comment by rajofearth** [comment]
> Not only Gemini 1.5 Pro but also Claude 3. I understand they support azure for open source models, but it should be on evreryones interest to be able to use more than one of the top closed source leaders.

Yeah rather than being focused on one of the top shelf models this project to should diverge a little towards other models and this is what will make this project stand out

**Comment by nimrod4278** [comment]
I think using something like liteLLM is best for this use case. anyway Gemini has now function calling, good results with json generation

**Reply by rajofearth** [reply]
Yeah, i recently show this been used in [open-interpreter](https://github.com/OpenInterpreter/open-interpreter?tab=readme-ov-file#change-your-language-model) project, this would solve the difficulties they are having.
